### DISCUSSION TITLE
alternate for Phalcon\Cache\Frontend\Data in phalcon 4

### DISCUSSION THREAD

**Discussion by Vermasj** [phalcon/cphalcon#15339] [Q&A]
I am upgrading form Phalcon 3.x to 4.x . I saw that class Phalcon\Cache\Frontend\Data was removed but I am unable to find the alternate class for that ..do I have to change the entire caching logic or is there a class that I can just use instead.

**Comment by quasipickle** [comment]
It looks like 4.0 moved away from discrete Front End and Back End classes.

https://docs.phalcon.io/4.0/en/cache

Your usage of the cache probably won't change, but your setup likely will.

**Comment by niden** [comment]
@Vermasj The changes you will need are really minimal. Actually it will be better for you, since instead of defining a front and a back cache, you now define only one service. The documentation has a pretty much copy paste example on how you can use this. You will however need to ensure that your interfaces are correct - if you use them or extend the cache class - because the new Cache object is PSR compliant.
